src/workerd/util/sqlite.h

```diff
@@ -19,6 +20,20 @@ namespace workerd {
 using kj::byte;
 using kj::uint;
 
+class SqliteObserver {
+public:
+  virtual void addRowsRead(uint64_t rowsRead) {}
+  virtual void addRowsWritten(uint64_t rowsWritten) {}
+  virtual void setSrsStoredBytesCallback(kj::Function<uint64_t()> callback) {}
+
+  virtual ~SqliteObserver() noexcept(false) {}
+
+  static SqliteObserver& getDefaultObserver() {
+    static SqliteObserver defaultObserver;
+    return defaultObserver;
+  }
```

Comment:
We normally just declare a constant for these default instances, like:

`static const SqliteObserver DEFAULT;`